### PR TITLE
Fix pacts

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

Pact tests are failing on the Provider side due to the UI making requests for fields that can be either strings or null, and Pact not handling these very well.

## Changes in this PR

- Don't expect an Earliest Release Date on an Organisation's referrals (this is only used in the Refer (`/me`) endpoint)
- Make first and last names not nullable (this will be updated on the API side)
- Ensure we don't set Earliest Release Dates on the Referral Summary factory without a Sentence
- Ignore `tasksCompleted` on Pact referral summaries - this shouldn't be sent by the API (at the moment)

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/@types/models/Referral.ts [hunk 1]
```diff
@@ -40,8 +40,8 @@ type ReferralSummary = {
   earliestReleaseDate?: Person['conditionalReleaseDate'] | Person['paroleEligibilityDate'] | Person['tariffDate']
   organisationId?: Organisation['id']
   prisonerName?: {
-    firstName?: Prisoner['firstName']
-    lastName?: Prisoner['lastName']
+    firstName: Prisoner['firstName']
+    lastName: Prisoner['lastName']
```
Comment (yndajas) [inline server/@types/models/Referral.ts:44]: These two `CaseListUtils` tests need removing with this commit: ``'when `prisonerName.firstName` is `undefined`'`` and ``when `'prisonerName.lastName` is `undefined`'``

**Review by yndajas — COMMENTED** [review]
Just needs one fix I think

File: server/@types/models/Referral.ts (hunk 1 above)
Comment (Gweaton) [reply server/@types/models/Referral.ts:44]: Yeah the pacts are currently passing on the API side so I thought I'd make the API changes before merging this one in, but yep, these tests need updating

Comment (Gweaton) [reply server/@types/models/Referral.ts:44]: Argh I take it back, a new deployment has randomised some data again - will need to sort this now

**Review by yndajas — APPROVED** [review]

**Pull request merged**